util/src/main/java/com/google/cloud/hadoop/util/HttpTransportFactory.java

```diff
@@ -165,4 +177,70 @@ static URI parseProxyAddress(@Nullable String proxyAddress) {
           String.format("Invalid proxy address '%s'.", proxyAddress), e);
     }
   }
+
+  /** Wrapper class to inject socketKeepAlive property while creating the socket */
+  static class ConfiguredSslSocketFactory extends SSLSocketFactory {
```

Comment:
If this is only not `private` to support testing, then please add the `@VisibleForTesting` annotation.

Reply:
Good point done.